src/main/java/org/spongepowered/api/plugin/PluginManager.java

```diff
@@ -77,4 +79,17 @@
      */
     boolean isLoaded(String id);
 
+    /**
+     * Lets the given plugin modify the world generator. Does nothing if the
+     * plugin does not implement the {@link WorldGeneratorProvider} interface.
+     *
+     * @param plugin The plugin to get the world generator from.
+     * @param worldName Name of the world.
+     * @param settings A string with (usually user-supplied) settings for the
+     *        world generator plugin, it's up to the world generator plugin to
+     *        do anything with this.
+     * @param worldGenerator The world generator to modify.
+     */
+    void modifyWorldGenerator(PluginContainer plugin, String worldName, String settings, WorldGenerator worldGenerator);
```

Comment:
This method was intended for inter-plugin communication. Let's say the command

`/mv create testworld normal -g AwesomeGenerator`

was called, then Multiverse would find the PluginContainer of the plugin AwesomeGenerator, use the above method to let it modify the WorldGenerator of the soon-to-be world, and then Multiverse would call server.createWorld with the modified WorldGenerator.

However, I think a better solution would be to add a method to #515's WorldBuilder class. Something like this?

```java
/**
 * Just before the actual world is created, each plugin listed here is notified so that it can
 * modify the world generator of the new world, for example to inject a custom biome
 * generator or populator.
 */
WorldBuilder plugins(PluginContainer... plugins)
```

I will make sure the plugin receives a DataContainer instead of a String.